cli/dts/lib.deno.unstable.d.ts

```diff
@@ -203,10 +203,67 @@ declare namespace Deno {
    */
   export function consoleSize(
     rid: number,
-  ): {
+  ): ConsoleSize;
+
+  /**
+   * **UNSTABLE**: New API, yet to be vetted.
+   *
+   * Sets the size of the console.
+   *
+   * ```ts
+   * Deno.resizeConsole(Deno.stdout.rid, {
+   *   rows: 20,
+   *   columns: 80,
+   * });
+   * ```
+   */
+  export function resizeConsole(
+    rid: number,
+    size: ConsoleSize,
+  ): void;
+
+  /**
+   * **UNSTABLE**: New API, yet to be vetted.
+   *
+   * Opens a new Pty interface.
+   *
+   * ```ts
+   * const pty = Deno.openPty({
+   *   rows: 20,
+   *   columns: 80,
+   * });
+   *
+   * await Deno.spawn(Deno.execPath(), {
+   *   args: [
+   *     "eval",
+   *     `
+   *       const size = Deno.consoleSize(Deno.stdout.rid);
+   *       console.log("%sx%s", size.columns, size.rows);
+   *     `
+   *   ],
+   *   tty: pty.rid,
+   * });
+   *
+   * const reader = pty.readable.getReader();
+   * const { value } = await reader.read();
+   * assertEquals(value, "20x80\n");
+   * Deno.close(pty.rid);
+   * ```
+   */
+  export function openPty(options: ConsoleSize): Pty;
+
+  /** A Pty fork with readable and writable streams. */
+  export interface Pty {
+    rid: number;
+    readable: ReadableStream<Uint8Array>;
+    writable: WritableStream<Uint8Array>;
+  }
```

Comment:
Add a `close()` method like other resources have
```suggestion
  export class Pty {
    readonly rid: number;
    readonly readable: ReadableStream<Uint8Array>;
    readonly writable: WritableStream<Uint8Array>;
    close(): void;
  }
```

Reply:
@nayeemrmn not all resources have that, ie the new subprocess API does not have a close method. also on that note, the `rid` should not be exposed either.

Reply:
Spawn has alternate ways of closing / being waited on, things that are closed primarily by `Deno.close(res.rid)` should have a wrapper method.

Hiding `rid`s (even if they're not used for normal API usage) might be a problem for metrics and resource leakage tracking, but I'm not sure if that's practically a problem especially with our better sanitizer messages. Sure, I agree with hiding `rid` in a weak map for now.

Reply:
> I agree with hiding `rid` in a weak map for now.

instead of a weakmap, it should be a symbol (as we do with web apis), for consistency.

overall, from what i remember discussing a few months back, we would want to move away from allowing manual resource management (so closing or resource and other usages of resource ids)

Reply:
@nayeemrmn I suppose that this method will use `core.tryClose()` under the hood?